### DISCUSSION TITLE
コードレビューのやり方

### DISCUSSION THREAD

**Discussion by pss-aileen** [first-contributions-ja/first-contributions-ja.github.io#40] [【運営】相談・報告]
- [チームでコードを書き始めた後、「どうやらレビューってやつをした方が良いらしい」くらいの若手に向けた資料です。](https://gist.github.com/taichi/b36c135001c161aeb1b240327ed044d6)

コードレビューやプルリク、
色々なことがはじめてなのでこういったことも話した方が良いかも？
次回のMTG時に共通認識が必要なものをまとめて話せれば良いなと思った。

**Comment by kazzyfrog** [comment]
ですね！
色々確認しつつ、まとめてMTGのアジェンダ的にしてもいいかも！

- コードレビュー
- プルリク
- イシュー

他にも、ドキュメントやイシューやディスカッションなど、どこをどの順番で見ればキャッチアップできるか、など？

**Comment by kazzyfrog** [comment]
コードレビューのやり方について、色々メモしていきますね！

- [レビューの大まかな流れ](https://developer.mamezou-tech.com/blogs/2022/06/29/reviewing-pr-on-github/)
- まず[セルフレビュー](https://beta-chelsea.hatenadiary.jp/entry/2020/12/19/125756)するのも大事！
- より実践的なベストプラクティスとして、こんな[まとめ記事](https://zenn.dev/s_masa/articles/956ec4c9814ddf)がある

**Reply by kazzyfrog** [reply]
オープンソースプロジェクトの場合、PRは仕事ではなく善意から来る貢献だったりするので、
できるだけフランクにディスカッションするのが大事ですよね！

一方で、[LGTM](https://qiita.com/e99h2121/items/072fa31827bf4591d799)についての記事で、結構良いヒントが書いてありました：
> レビュアーは「LGTM」が「このコードは我々の基準を満たしている」と確信できるように、レビューに十分な時間を費やすことが重要です。しかし個々のレスポンスはやはり理想的には迅速であるべきです。

PRを承認するかどうかは、コードやその他ドキュメントに対する自分たちの基準が大事になってきそうです！

**Reply by kazzyfrog** [reply]
@pss-aileen 
もし良かったら、 #43 を練習がてらレビューしてみてください！
わざとtypoしてますw
あと、PRの内容も何やら悩ましい問題を抱えています！

レビューは、変更されたファイルの、気になる行を＋マークでコメント追加できます！
詳しくは、上に貼った「レビューの大まかな流れ」を参照してください！

もしわからない事があれば、サポートしますよ🙌

**Reply by pss-aileen** [reply]
わー！ありがとうございます！！
ちょっとたてこんでいるので今日夜、明日のどこかで確認させていただきます🙇
困ったら相談させてください！！
